src/task.c

```diff
@@ -50,100 +40,71 @@ task_init_tid_bitmask(uint64 *tid_bitmask)
    }
 }
 
+static inline uint64 *
+task_system_get_tid_bitmask(task_system *ts)
+{
+   return &ts->tid_bitmask;
+}
+
+static threadid *
+task_system_get_max_tid(task_system *ts)
+{
+   return &ts->max_tid;
+}
+
 /*
- * task_init_threadid() - Initialize a new thread's thread ID (index).
+ * Return the bitmasks of tasks active. Mainly intended as a testing hook.
+ */
+uint64
+task_active_tasks_mask(task_system *ts)
+{
+   return *task_system_get_tid_bitmask(ts);
+}
 
- * The first thing a newly started thread that wants to use Splinter is expected
- * to do is to "register itself" with the task system. This function is
- * defined as a thread's hook and gets run as part of the thread's
- * registration. This function initializes each thread's thread-ID, which
- * is really an index into the array of thread-specific data structures.
+/*
+ * Allocate a threadid
  */
-static void
-task_init_threadid(task_system *ts)
+static threadid
+allocate_threadid(task_system *ts)
 {
-   threadid  tid         = INVALID_TID;
-   uint64   *tid_bitmask = task_system_get_tid_bitmask(ts);
-   threadid *max_tid     = task_system_get_max_tid(ts);
+   threadid  tid        = INVALID_TID;
+   uint64   *tid_bitset = task_system_get_tid_bitmask(ts);
+   threadid *max_tid    = task_system_get_max_tid(ts);
```

Comment:
As this variable is not used till much later, for readability, consider moving this line to just before L95 below.